src/Providers/ProviderRegistry.php

```diff
@@ -66,6 +83,27 @@ public function registerProvider(string $className): void
             );
         }
 
+        // If there is already a HTTP transporter instance set, hook it up to the provider as needed.
+        try {
+            $httpTransporter = $this->getHttpTransporter();
+            $this->setHttpTransporterForProvider($className, $httpTransporter);
+        } catch (RuntimeException $e) {
+            // Ignore.
+        }
```

Comment:
Per our chat, let's update the comment here to make it clear that we're avoiding a race condition here, and also setting the transporter later updates all the providers.